### ISSUE TITLE
Release v0.3.0

### ISSUE THREAD

**Issue by danielvegamyhre** [kubernetes-sigs/jobset#347]
## Release Checklist

- [x] All [OWNERS](https://github.com/kubernetes-sigs/jobset/blob/main/OWNERS) must LGTM the release proposal
- [x] Verify that the changelog in this issue is up-to-date
- [x] For major or minor releases (v$MAJ.$MIN.0), create a new release branch.
  - [x] an OWNER creates a vanilla release branch with
        `git branch release-$MAJ.$MIN main`
  - [x] An OWNER pushes the new release branch with
        `git push release-$MAJ.$MIN`
- [x] Update things like README, deployment templates, docs, configuration, test/e2e flags.
      Submit a PR against the release branch: 
- [x] An OWNER [prepares a draft release](https://github.com/kubernetes-sigs/jobset/releases)
  - [x] Write the change log into the draft release.
  - [x] Run
      `make artifacts IMAGE_REGISTRY=registry.k8s.io/jobset GIT_TAG=$VERSION`
      to generate the artifacts and upload the files in the `artifacts` folder
      to the draft release.
- [x] An OWNER creates a signed tag running
     `git tag -s $VERSION`
      and inserts the changelog into the tag description.
      To perform this step, you need [a PGP key registered on github](https://docs.github.com/en/authentication/managing-commit-signature-verification/checking-for-existing-gpg-keys).
- [x] An OWNER pushes the tag with
      `git push $VERSION`
  - Triggers prow to build and publish a staging container image
      `gcr.io/k8s-staging-jobset/jobset:$VERSION`
- [x] Submit a PR against [k8s.io](https://github.com/kubernetes/k8s.io), 
      updating `k8s.gcr.io/images/k8s-staging-jobset/images.yaml` to
      [promote the container images](https://github.com/kubernetes/k8s.io/tree/main/k8s.gcr.io#image-promoter)
      to production: 
- [x] Wait for the PR to be merged and verify that the image `registry.k8s.io/jobset/jobset:$VERSION` is available.
- [x] Publish the draft release prepared at the [Github releases page](https://github.com/kubernetes-sigs/jobset/releases).
- [x] Add a link to the tagged release in this issue: 
- [x] Send an announcement email to `sig-apps@kubernetes.io`, `sig-scheduling@kubernetes.io` and `wg-batch@kubernetes.io` with the subject `[ANNOUNCE] JobSet $VERSION is released`
- [x] Add a link to the release announcement in this issue: 
- [x] For a major or minor release, update `README.md` and `docs/setup/install.md`
      in `main` branch: 
- [x] For a major or minor release, create an unannotated _devel_ tag in the
      `main` branch, on the first commit that gets merged after the release
       branch has been created (presumably the README update commit above), and, push the tag:
      `DEVEL=v0.$(($MAJ+1)).0-devel; git tag $DEVEL main && git push $DEVEL`
      This ensures that the devel builds on the `main` branch will have a meaningful version number.
- [x] Close this issue

## Changelog
* Automatically move generated client-go files to repo root by @danielvegamyhre in https://github.com/kubernetes-sigs/jobset/pull/221
* use int32 rather than int by @kannon92 in https://github.com/kubernetes-sigs/jobset/pull/220
* Use get_namespaced_custom_object for getting jobset by @danielvegamyhre in https://github.com/kubernetes-sigs/jobset/pull/225
* Add pyproject.toml and cleanup sdk installation by @kannon92 in https://github.com/kubernetes-sigs/jobset/pull/223
* Bump github.com/open-policy-agent/cert-controller from 0.7.1-0.20230527041600-d09f1db9c066 to 0.8.0 by @dependabot in https://github.com/kubernetes-sigs/jobset/pull/229
* Bump sigs.k8s.io/structured-merge-diff/v4 from 4.2.3 to 4.3.0 by @dependabot in https://github.com/kubernetes-sigs/jobset/pull/230
* Remove generate target from build, run and test targets by @a-hilaly in https://github.com/kubernetes-sigs/jobset/pull/228
* Bump k8s.io/client-go from 0.27.3 to 0.27.4 by @dependabot in https://github.com/kubernetes-sigs/jobset/pull/232
* Bump k8s.io/code-generator from 0.27.3 to 0.27.4 by @dependabot in https://github.com/kubernetes-sigs/jobset/pull/233
* Bump github.com/onsi/gomega from 1.27.8 to 1.27.10 by @dependabot in https://github.com/kubernetes-sigs/jobset/pull/238
* add kep template by @kannon92 in https://github.com/kubernetes-sigs/jobset/pull/242
* cover some unit test gaps in jobset_webhook.go by @kannon92 in https://github.com/kubernetes-sigs/jobset/pull/245
* add an example for success policy by @kannon92 in https://github.com/kubernetes-sigs/jobset/pull/243
* add patch rbac for events by @kannon92 in https://github.com/kubernetes-sigs/jobset/pull/248
* Bump sigs.k8s.io/controller-runtime from 0.15.0 to 0.15.1 by @dependabot in https://github.com/kubernetes-sigs/jobset/pull/249
* Fix generation by @kannon92 in https://github.com/kubernetes-sigs/jobset/pull/251
* boiler plate code for kep templating by @kannon92 in https://github.com/kubernetes-sigs/jobset/pull/252
* pointer to ptr conversion by @kannon92 in https://github.com/kubernetes-sigs/jobset/pull/255
* add golangci and fix some minor items by @kannon92 in https://github.com/kubernetes-sigs/jobset/pull/254
* Add source to pytorch mnist example by @danielvegamyhre in https://github.com/kubernetes-sigs/jobset/pull/256
* Update installation docs for v0.2.1 by @danielvegamyhre in https://github.com/kubernetes-sigs/jobset/pull/264
* Fail fast if we cannot build indexes by @danielvegamyhre in https://github.com/kubernetes-sigs/jobset/pull/265
* Add update/patch rbac verbs for resource jobs/status by @danielvegamyhre in https://github.com/kubernetes-sigs/jobset/pull/268
* Bump k8s.io/api from 0.27.4 to 0.27.5 by @dependabot in https://github.com/kubernetes-sigs/jobset/pull/272
* Bump k8s.io/code-generator from 0.27.4 to 0.27.5 by @dependabot in https://github.com/kubernetes-sigs/jobset/pull/273
* Bump github.com/onsi/ginkgo/v2 from 2.11.0 to 2.12.0 by @dependabot in https://github.com/kubernetes-sigs/jobset/pull/274
* Remove incompatible shuffle option from cuda args by @danielvegamyhre in https://github.com/kubernetes-sigs/jobset/pull/287
* Bump k8s.io/client-go from 0.27.4 to 0.27.5 by @dependabot in https://github.com/kubernetes-sigs/jobset/pull/288
* Update docs to use latest release by @danielvegamyhre in https://github.com/kubernetes-sigs/jobset/pull/293
* Add validation enforcing that network subdomain adheres to RFC 1035 by @danielvegamyhre in https://github.com/kubernetes-sigs/jobset/pull/278
* add a tensorflow example and update task documentation by @kannon92 in https://github.com/kubernetes-sigs/jobset/pull/253
* K8 minor bump by @kannon92 in https://github.com/kubernetes-sigs/jobset/pull/301
* Bump k8s.io/client-go from 0.27.5 to 0.27.6 by @dependabot in https://github.com/kubernetes-sigs/jobset/pull/300
* Validate the generated job names will be dns 1035 compliant by @danielvegamyhre in https://github.com/kubernetes-sigs/jobset/pull/284
* Add troubleshooting docs by @danielvegamyhre in https://github.com/kubernetes-sigs/jobset/pull/303
* upgrade to k8s 1.28 by @kannon92 in https://github.com/kubernetes-sigs/jobset/pull/306
* Bump github.com/onsi/ginkgo/v2 from 2.12.0 to 2.12.1 by @dependabot in https://github.com/kubernetes-sigs/jobset/pull/308
* Update troubleshooting docs by @danielvegamyhre in https://github.com/kubernetes-sigs/jobset/pull/307
* Add node selector strategy for exclusive job placement per topology by @danielvegamyhre in https://github.com/kubernetes-sigs/jobset/pull/311
* Bump github.com/onsi/gomega from 1.27.10 to 1.28.0 by @dependabot in https://github.com/kubernetes-sigs/jobset/pull/313
* Bump github.com/open-policy-agent/cert-controller from 0.9.0 to 0.10.0 by @dependabot in https://github.com/kubernetes-sigs/jobset/pull/314
* Bump github.com/onsi/ginkgo/v2 from 2.12.1 to 2.13.0 by @dependabot in https://github.com/kubernetes-sigs/jobset/pull/316
* Bump github.com/google/go-cmp from 0.5.9 to 0.6.0 by @dependabot in https://github.com/kubernetes-sigs/jobset/pull/315
* Fix typo in API comments by @danielvegamyhre in https://github.com/kubernetes-sigs/jobset/pull/317
* Bump github.com/onsi/gomega from 1.28.0 to 1.28.1 by @dependabot in https://github.com/kubernetes-sigs/jobset/pull/320
* Bump sigs.k8s.io/controller-runtime from 0.16.2 to 0.16.3 by @dependabot in https://github.com/kubernetes-sigs/jobset/pull/322
* Bump github.com/onsi/gomega from 1.28.1 to 1.29.0 by @dependabot in https://github.com/kubernetes-sigs/jobset/pull/323
* Bump sigs.k8s.io/structured-merge-diff/v4 from 4.3.0 to 4.4.1 by @dependabot in https://github.com/kubernetes-sigs/jobset/pull/326
* Parallelize job creation by @danielvegamyhre in https://github.com/kubernetes-sigs/jobset/pull/328
* Self-nominate kannon92 for a reviewer for jobset. by @kannon92 in https://github.com/kubernetes-sigs/jobset/pull/327
* Bump github.com/onsi/gomega from 1.29.0 to 1.30.0 by @dependabot in https://github.com/kubernetes-sigs/jobset/pull/330
* Migrate to use kube_codegen.sh to generate client by @ashnehete in https://github.com/kubernetes-sigs/jobset/pull/329
* Bump github.com/onsi/ginkgo/v2 from 2.13.0 to 2.13.1 by @dependabot in https://github.com/kubernetes-sigs/jobset/pull/331
* Bump k8s.io/client-go from 0.28.3 to 0.28.4 by @dependabot in https://github.com/kubernetes-sigs/jobset/pull/337
* Add pod webhook + reconciler for new "leader pod" exclusive job placement strategy by @danielvegamyhre in https://github.com/kubernetes-sigs/jobset/pull/309
* Rename shared pkg to placement by @danielvegamyhre in https://github.com/kubernetes-sigs/jobset/pull/340
* add flags for qps and burst instead of hardcoding them by @dejanzele in https://github.com/kubernetes-sigs/jobset/pull/343
* Bump github.com/onsi/ginkgo/v2 from 2.13.1 to 2.13.2 by @dependabot in https://github.com/kubernetes-sigs/jobset/pull/345
* allow for podman to be used for running tests by @kannon92 in https://github.com/kubernetes-sigs/jobset/pull/333
* Update pod reconciler to handle race conditions in exclusive placement by @danielvegamyhre in https://github.com/kubernetes-sigs/jobset/pull/342
* [release-0.3 branch] Update with installation docs with v0.3.0 by @danielvegamyhre in https://github.com/kubernetes-sigs/jobset/pull/348

**Comment by kannon92** [comment]
What do you want to do about https://github.com/kubernetes-sigs/jobset/issues/239?

**Comment by danielvegamyhre** [comment]
Woops, forgot I made a release issue already, thanks!

**Comment by danielvegamyhre** [comment]
@kannon92 we'll include the 2 outstanding "must haves" in the next release

**Comment by kannon92** [comment]
> @kannon92 we'll include the 2 outstanding "must haves" in the next release

That sounds good to me!

**Comment by danielvegamyhre** [comment]
I probably should have put those 2 in "nice to haves" anyway since there is no customer requiring these features currently, so they aren't really hard requirements, it was just a goal to include them in this release, but other things ended up taking priority/time away from them. I will move them to nice to haves before cutting the release to document this.

**Comment by danielvegamyhre** [comment]
Release v0.3.0 published: https://github.com/kubernetes-sigs/jobset/releases/tag/v0.3.0

**Comment by danielvegamyhre** [comment]
Release announcement email: https://groups.google.com/a/kubernetes.io/g/wg-batch/c/Jt0M3yE4S0M/m/eQVcPPR1AAAJ
